bids/layout/tests/test_layout.py

```diff
@@ -471,6 +471,60 @@ def test_parse_file_entities_from_layout(layout_synthetic):
     target = {'desc': 'bleargh'}
     assert target == layout.parse_file_entities(filename, config='derivatives')
 
+def test_parse_file_entities_from_layout_cached_db(layout_synthetic_cached_db):
+    layout = layout_synthetic_cached_db
+    filename = '/sub-03_ses-07_run-4_desc-bleargh_sekret.nii.gz'
+
+    # Test with entities taken from bids config
+    target = {'subject': '03', 'session': '07', 'run': 4, 'suffix': 'sekret',
+              'extension': 'nii.gz'}
+    assert target == layout.parse_file_entities(filename, config='bids')
+    config = Config.load('bids')
+    assert target == layout.parse_file_entities(filename, config=[config])
+    assert target == layout.parse_file_entities(filename, scope='raw')
+
+    # Test with default scope--i.e., everything
+    target = {'subject': '03', 'session': '07', 'run': 4, 'suffix': 'sekret',
+              'desc': 'bleargh', 'extension': 'nii.gz'}
+    assert target == layout.parse_file_entities(filename)
+    # Test with only the fmriprep pipeline (which includes both configs)
+    assert target == layout.parse_file_entities(filename, scope='fmriprep')
+    assert target == layout.parse_file_entities(filename, scope='derivatives')
+
+    # Test with only the derivative config
+    target = {'desc': 'bleargh'}
+    assert target == layout.parse_file_entities(filename, config='derivatives')
+
+
+def test_parse_file_entities_from_layout_cached_db_replay(layout_synthetic_cached_db_replay):
```

Comment:
We use py.test for testing, so I think this test could be removed in favor of [parameterization](https://docs.pytest.org/en/latest/parametrize.html) of the previous one... i.e., I think you should be able to just decorate the preceding function with `@pytest.mark.parametrize(layout=[layout_synthetic_cached_db, layout_synthetic_cached_db_replay])`, no need for duplicate code.

Reply:
@tyarkoni I am not all that familiar with py.test.  I do not have time to address the improvements you suggest for testing.  Could you please implement your suggestion as a separate PR?

I really need this code contained in this PR merged so that I can get the other PR's ready for submission, and so that I can get back to my primary data analysis tasks.